widget.html

```diff
@@ -1,121 +1,139 @@
-                                                <th scope="col" style="text-align: right;"><input name="comments9" size="35" type="text" /></th>
-                                </tr>
-                                <tr>
-                                                <th scope="col" style="text-align: left; width: 35px;"><input name="substance10" size="35" type="text" /></th>
-                                                <td scope="col" style="text-align: right; width: 8px;"><input name="amount10_number" size="8" type="text" /></td>
-                                                <th scope="col" style="text-align: right; width: 8px;"><input name="fw10_number" size="8" type="text" /></th>
-                                                <th scope="col" style="text-align: right; width: 8px;"><input name="moles10_formula" size="8" type="text" value="(#{moles1_formula}*#{equivalents10_number}).toFixed(2)" /></th>
-                                                <th scope="col" style="text-align: right; width: 5px;"><input name="equivalents10_number" size="5" type="text" /></th>
-                                                <th scope="col" style="text-align: right; width: 8px;"><input name="volume10_formula" size="8" type="text" value="(#{amount10_number}/#{density10_number}).toFixed(2)" /></th>
-                                                <th scope="col" style="text-align: right; width: 5px;"><input name="density10_number" size="5" type="text" /></th>
-                                                <th scope="col" style="text-align: right;"><input name="comments10" size="35" type="text" /></th>
-                                </tr>
-                </tbody>
-</table>
\ No newline at end of file
+<style type="text/css">table {
+    border-collapse: collapse;
+    border: 3px solid purple;;
+    cellpadding: 1px;
+    cellspacing: 1px;
+    width: 930px;
+    height: 100px;
+    font-family: Times;
+}
+
+input[type=text]:enabled {
+    background: #FFFFFF;
+}
+
+input[type=text]:disabled {
+    background: #dddddd;
+}
+</style>
+<table id="T01">
+    <thead>
+    <tr>
+        <th scope="col" style="width: 35px;"><strong>Substance</strong></th>
+        <th scope="col" style="width: 8px;"><strong>Amount<br />
+            (mg)</strong></th>
+        <th scope="col" style="width: 8px;"><strong>FW<br />
+            (g/mol)</strong></th>
+        <th scope="col" style="width: 8px;"><strong>Moles<br />
+            (mmol)</strong></th>
+        <th scope="col" style="width: 5px;"><strong>Equiv.</strong></th>
+        <th scope="col" style="width: 8px;"><strong>Volume<br />
+            (uL)</strong></th>
+        <th scope="col" style="width: 5px;"><strong>Density<br />
+            (g/mL)</strong></th>
+        <th scope="col" style="width: 35px;"><strong>Comments</strong></th>
+    </tr>
+    </thead>
+    <tbody>
+    <tr class="initialRow">
```

Comment:
you should not need to label the initial row by the way. the selector `:first-child` should be fine under `tbody`.